crates/project-model/src/workspace.rs

```diff
@@ -1448,8 +1451,9 @@ fn add_target_crate_root(
         Arc::new(cfg_options),
         potential_cfg_options.map(Arc::new),
         env,
-        matches!(kind, TargetKind::Lib { is_proc_macro: true }),
         origin,
+        matches!(kind, TargetKind::Lib { is_proc_macro: true }),
+        Some(pkg.manifest.parent().to_path_buf()),
```

Comment:
@Veykril is this ok to do here?

Reply:
That unconditionally sets it to the manifest dir which is incorrect, as was  discussed, for workspace members this needs to be the workspace root

Reply:
sorry about that, fixing